MobileApplication/MobileApplication/ViewModels/GoalsViewModel.cs

```diff
@@ -76,13 +76,20 @@ private async void OnAddItem(object obj)
             await Shell.Current.GoToAsync(nameof(NewItemPage));
         }
 
-        async void OnItemSelected(Goal item)
+        async void OnItemSelected(Goal goal)
         {
-            if (item == null)
+            if (goal == null)
                 return;
 
             // This will push the ItemDetailPage onto the navigation stack
-            await Shell.Current.GoToAsync($"{nameof(ItemDetailPage)}?{nameof(ItemDetailViewModel.ItemId)}={item.Id}");
+            Item item = new Item(goal.Id.ToString(), goal.Title,
+                                string.Concat(goal.Amount.ToString(), '\n',
+                                goal.Description, '\n',
+                                goal.Creationdate.ToString(), '\n',
+                                goal.Deadlinedate.ToString()));
```

Comment:
The same here should be Goal object, not Item.

Reply:
So there should be different itemDetailPages for every view model?